### DISCUSSION TITLE
是否可通过一些脚本，使候选窗口成为交互性更强的窗口

### DISCUSSION THREAD

**Discussion by WhiteTiger-WT** [rime/home#1155] [Ideas]
1. 如通过lua实现计算器功能，引导输入算式，候选出计算结果。如输入`C2+3*2/(3-1)`，引导候选出 `5`。
2. 将本地某应用路径写入，通过候选项，打开本地应用。
3. 成对符号输入时如`〔〕`，光标自动转到中间
4. 是否有获取最近上屏的内容的api，可通过 lua 加工最近上屏内容后输出，使候选内容具有动态性。如输入：`10`，引导候选可为`10 是整数`；上屏 `李明`，候选出 `姓名，李明 `。~~通过 history_translator 能实现这个吗？~~
5. 通过指定按键可撤消上次上屏内容。如上屏`马到功成`，通过按键，撤消这次上屏。（这与`Ctrl+z`有相似作用，不完全相同。另`Ctrl+z`左手要完全离开主键盘区，不便。希望可以实现自定义按键实现该功能，如 `;u` 实现撤消）
6. 通过候选项内容打开浏览器并搜索指定内容。可基于第4点，搜索最近上屏词。如上屏 `三体` 后，再输引导编码候选出 `1. [百度]：三体` `2. [bilibili]：三体` ，选相应候选，则跳转到网站搜索。

**Comment by LEOYoon-Tsaw** [comment]
您另做一App吧，這已經不是輸入法了。

**Reply by WhiteTiger-WT** [reply]
多多平台除了第 1 点和第 4 点，都可以实现。那些功能还是比较方便的。

**Reply by LEOYoon-Tsaw** [reply]
輸入打車能自動打車就更方便了
> 多多平台除了第 1 点和第 4 点，都可以实现。那些功能还是比较方便的。

**Reply by WhiteTiger-WT** [reply]
计算器功能
https://github.com/baopaau/rime-lua-collection/blob/master/calculator_translator.lua
